File: 383-ransom-note/ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        seen = {}

        for i in range(len(magazine)):
            seen[magazine[i]] = seen.get(magazine[i], 0) + 1

        for i in range(len(ransomNote)):
            if ransomNote[i] not in seen or seen[ransomNote[i]] == 0:
                return False

            seen[ransomNote[i]] -= 1

        return True


            

File: 383-ransom-note/ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map <char,int> seen;
        for (char c : magazine){
            seen[c]++;
        }
        for (char c : ransomNote){
            seen[c]--;
        }
        for (auto x : seen){
            if (x.second < 0){
                return false;
            }
        }
        return true;
    }
};